engine/src/main/java/org/hibernate/search/engine/search/predicate/dsl/SearchPredicateFactory.java

```diff
@@ -82,6 +83,59 @@ public interface SearchPredicateFactory {
 	@Deprecated
 	PredicateFinalStep bool(Consumer<? super BooleanPredicateClausesStep<?>> clauseContributor);
 
+	/**
+	 * Match documents if they match a conjunction of bcolean clauses.
+	 *
+	 * @return The initial step of a DSL where the "AND" predicates can be defined.
+	 * @see AndPredicateOptionsStep
+	 */
+	AndPredicateOptionsStep<?> and();
+
+	/**
+	 * Match documents if they match a conjunction of previously-built {@link SearchPredicate}.
+	 *
+	 * @return The step of a DSL where additional "AND" predicates can be defined and options be set.
+	 * @see AndPredicateOptionsStep
+	 */
+	AndPredicateOptionsStep<?> and(
+			SearchPredicate firstSearchPredicate,
+			SearchPredicate... otherSearchPredicates);
```

Comment:
This will allow things such as:

```
f.and( e1, e2 ).add( e3 )
```

I think it's a bit odd?

I'd rather promote consistent syntax, in this case either:

```
f.and( e1, e2, e3 )
```

... or:


```
f.and().add( e1 ).add( e2 ).add( e3 )
```

... but never mixed like in the first example.

Another argument against allowing this is code completion in the IDE: users writing `.and( e1, e2 )` and then using auto-completion to see what's possible on the returned value will see the `with` and `add` methods, which could be confusing. I'd rather not expose them to that, especially considering that we're trying to introduce a simpler API here?

To that end, you could introduce a `AndPredicateClausesStep`, which extends `AndPredicateOptionsStep`, and push all `add`/`with` methods there. `and()` would return `AndPredicateClausesStep`, which allows `add`/`constantScore`/`boost`/`toPredicate` but `and(...)` would return `AndPredicateOptionsStep`, which only allows `constantScore`/`boost`/`toPredicate`.

WDYT?

(Same idea for `or`, of course)